### PR TITLE
BeVigil Source Added

### PR DESCRIPTION
**Pull request by alt-glitch**
BeVigil scans mobile applications and makes OSINT data available for security researchers.
Let me know if the code can be improved a bit :))

### CODE REVIEW

**Comment by L1ghtn1ng** [comment]
@alt-glitch Thank your for your request, you have included your api key so that needs removing and your will want to revoke that key as its now compromised due to the accidental inclusion of it. You will want to remove it and see if if you can remove it from git history. Unfortunately I do not know how to do that so cannot help on that one.

File: theHarvester/discovery/bevigil.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    async def get_hostnames(self) -> set:
+        return self.totalhosts
+
+    async def get_interestingurls(self) -> set:
+        return self.interestingurls
+
+    async def process(self, proxy=False):
+        self.proxy = proxy
+        await self.do_search()
```
Comment (L1ghtn1ng) [inline theHarvester/discovery/bevigil.py:37]:
```suggestion
        await self.do_search()
        
```
This makes it compliant with flake8

**Comment by alt-glitch** [comment]
Hi
Yes I think I mistakenly saw the `api_keys.yaml` file in the `.gitignore` so thought it wouldn't get added. I realised my mistake and have revoked the API key. Thank you for the heads up!

File: theHarvester/discovery/bevigil.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        self.interestingurls = set()
+        self.key = Core.bevigil_key()
+        self.proxy = False
+
+    async def do_search(self):
+        subdomainEndpoint = f"https://osint.bevigil.com/api/{self.word}/subdomains/"
+        urlEndpoint = f"https://osint.bevigil.com/api/{self.word}/urls/"
+        headers = {
+            'X-Access-Token': self.key
+        }
```
Comment (L1ghtn1ng) [inline theHarvester/discovery/bevigil.py:17]:
```suggestion
        headers = {'X-Access-Token': self.key}
```
Just to remove un-needed space being taken up

**Review by L1ghtn1ng — CHANGES_REQUESTED** [review]
Only a few things that need slightly changing but looks good :+1:

File: theHarvester/discovery/bevigil.py (hunk 1 above)
Comment (alt-glitch) [reply theHarvester/discovery/bevigil.py:37]: This seems to be the same? Am i missing something ? :)

**Comment by L1ghtn1ng** [comment]
Thank you for adding this :+1:

**Pull request merged**